### PR TITLE
Fix tpaccept world teleport permissions

### PR DESCRIPTION
**Pull request by mind04**
### Details
Fix a world teleport permission bypass. A user with permission for a word is able to provide access to that world to users without permission. 

**Environments tested:**    

OS: Debian 10

Java version:  17

- [x] Most recent Paper version (1.XX.Y, git-Paper-BUILD)
- [ ] CraftBukkit/Spigot/Paper 1.12.2
- [ ] CraftBukkit 1.8.8

**Demonstration:** 
user1 has permission to world X
user2 has no permission to world X

With use of tpahere:
user1 goes to world X and executes /tpahere
user2 executes /tpaccept -> tp of user2 to world X

With use of tpa:
Both users are in the same world.
user2 executes /tpa user1
user1 teleports to word X
user1 executes /tpaccept -> tp of user2 to world X

### CODE REVIEW

**Review by JRoy — DISMISSED** [review]

**Comment by JRoy** [comment]
would like @mdcfe's or @pop4959's thoughts

**Review by pop4959 — DISMISSED** [review]
LGTM. Line 90 seems to be a regression from #3801, and the second permission check on line 94 was added in that same PR, which should follow logically that the condition is backwards currently there as well.

**Comment by pop4959** [comment]
AKA JRoy what were you thinking :)

**Comment by JRoy** [comment]
@mind04 Please update the pull request with 2.x or allow maintainer edits from pull requests.

**Comment by mind04** [comment]
Done

**Review by pop4959 — APPROVED** [review]

**Review by mdcfe — CHANGES_REQUESTED** [review]
Thanks for spotting this!

However, I feel these conditions should be pulled out into separate statements. The reason this slipped through in the first place is because these giant chained conditions are impossible to read at a glance. It would be nice to make the logic here a bit more readable.

Will follow up this review with suggested changes in a moment.

**Comment by mind04** [comment]
> Will follow up this review with suggested changes in a moment.

Any progress?

Is it an option to merge the pull as is? It is fixing a real issue, does not increase complicity and does not prevent a clean-up at a later date.

**Comment by mdcfe** [comment]
Hey, we did discuss this internally but apparently I forgot to respond here - apologies for that.

I did work on refactoring the class somewhat, the results of which can be seen here: https://github.com/EssentialsX/Essentials/commit/ff7569c77daa37664b15b64f70b3744898a465d3

I don't think we came to a decision internally at the time, but I would be happy to merge with the above changes, pending @pop4959 or @JRoy double checking it.

**Comment by pop4959** [comment]
@mind04 Apologies for taking so long to get back to you on this PR. It looks like it's been automatically closed due to the repository being deleted. Just want to let you know in case this wasn't intentional.

**Pull request closed without merging**
